src/pages/task/Metadata.tsx

```diff
@@ -33,57 +30,69 @@ interface Props {
 }
 
 export const Metadata: React.FC<Props> = ({ loading, data, error, taskId }) => {
-  const task = data ? data.task : null;
+  const task = data?.task ?? ({} as GetTaskQuery["task"]);
   const taskAnalytics = useTaskAnalytics();
 
-  const status = task?.status;
-  const spawnHostLink = task?.spawnHostLink;
-  const ingestTime = task?.ingestTime;
-  const finishTime = task?.finishTime;
-  const hostId = task?.hostId;
-  const startTime = task?.startTime;
-  const estimatedStart = task?.estimatedStart;
-  const timeTaken = task?.timeTaken;
-  const revision = task?.revision ?? "";
-  const baseCommit = revision.slice(0, 10);
-  const reliesOn = task?.reliesOn;
-  const baseTaskMetadata = task?.baseTaskMetadata;
-  const ami = task?.ami;
-  const distroId = task?.distroId;
-  const baseTaskDuration = baseTaskMetadata?.baseTaskDuration;
-  const baseTaskLink = baseTaskMetadata?.baseTaskLink;
-  const priority = task?.priority;
-  const patchMetadata = task?.patchMetadata;
-  const author = patchMetadata?.author;
+  const {
+    status,
+    spawnHostLink,
+    ingestTime,
+    finishTime,
+    hostId,
+    startTime,
+    estimatedStart,
+    timeTaken,
+    revision,
+    reliesOn,
+    baseTaskMetadata,
+    ami,
+    distroId,
+    priority,
+    patchMetadata,
+    buildVariant,
+    details,
+    generatedBy,
+    generatedByName,
+    minQueuePosition: taskQueuePosition,
+  } = task;
 
-  const details = task?.details;
+  const baseCommit = revision?.slice(0, 10);
+
+  const { baseTaskDuration, baseTaskLink } = baseTaskMetadata ?? {};
+
+  const { author, patchID } = patchMetadata ?? {};
   const oomTracker = details?.oomTracker;
 
   const hostLink = `${paths.host}/${hostId}`;
   const distroLink = `${getUiUrl()}/distros##${distroId}`;
 
-  const { data: taskQueuePositionData } = useQuery<
-    TaskQueuePositionQuery,
-    TaskQueuePositionQueryVariables
-  >(TASK_QUEUE_POSITION, { variables: { taskId } });
-
-  const taskQueuePosition = taskQueuePositionData?.task?.minQueuePosition ?? 0;
-
   return (
     <>
       <MetadataCard error={error} loading={loading} title="Task Metadata">
+        <P2 data-cy="task-metadata-build-variant">
+          Build Variant Name:{" "}
+          <StyledRouterLink
+            data-cy="build-variant-link"
+            to={`${getVersionRoute(patchID)}?page=0&variant=${buildVariant}`}
```

Comment:
`page` and `variant` could be added to the `getVersionRoute` function. Inputting them manually here has a slight code smell.

Reply:
Great catch i've also cleaned up other areas where we do this